services/smtp/smtpEmailService_test.go

```diff
@@ -0,0 +1,265 @@
+	testURI, _ := common.NewURIFromString("test")
+
+	setup.mockSMTPClient.EXPECT().SendEmail(fmt.Sprintf("%s:%s", testServer, testPort), testAuth,
+		testCfg.Email.NoreplyEmailAddr, []string{"rob@test.com"}, gomock.Any()).Return(nil).Times(1)
+	setup.mockTimeProvider.EXPECT().Now().Return(time.Unix(1, 0)).Times(1)
+	setup.mockAuthorizer.EXPECT().CreateServiceToken(setup.testCtx, testUserId, []common.UniformResourceIdentifier{testURI}, int64(1001)).
+		Return("", nil).Times(1)
+
+	err := setup.emailService.SendEmailVerificationEmail(setup.testCtx, entities.User{
+		ID:    testUserId,
+		Email: "rob@test.com",
+	}, []common.UniformResourceIdentifier{testURI})
+	assert.NoError(t, err)
+}
+
+func Test_SendEmailVerificationEmail__should_return_error_when_authorizer_returns_error(t *testing.T) {
+	setup := setupEmailTest(t)
+	defer setup.ctrl.Finish()
+	testURI, _ := common.NewURIFromString("test")
+
+	setup.mockTimeProvider.EXPECT().Now().Return(time.Unix(1, 0)).Times(1)
+	setup.mockAuthorizer.EXPECT().CreateServiceToken(setup.testCtx, testUserId, []common.UniformResourceIdentifier{testURI}, int64(1001)).
+		Return("", errors.New("authorizer err")).Times(1)
+
+	err := setup.emailService.SendPasswordResetEmail(setup.testCtx, entities.User{
+		ID: testUserId,
+	}, []common.UniformResourceIdentifier{testURI})
+
+	assert.Error(t, err)
+}
+
+func Test_SendPasswordResetEmail__should_not_return_error_when_sending_email_is_successful(t *testing.T) {
+	setup := setupEmailTest(t)
+	defer setup.ctrl.Finish()
+	testURI, _ := common.NewURIFromString("test")
+
+	setup.mockSMTPClient.EXPECT().SendEmail(fmt.Sprintf("%s:%s", testServer, testPort), testAuth,
+		testCfg.Email.NoreplyEmailAddr, []string{"rob@test.com"}, gomock.Any()).Return(nil).Times(1)
+	setup.mockTimeProvider.EXPECT().Now().Return(time.Unix(1, 0)).Times(1)
+	setup.mockAuthorizer.EXPECT().CreateServiceToken(setup.testCtx, testUserId, []common.UniformResourceIdentifier{testURI}, int64(1001)).
+		Return("", nil).Times(1)
+
+	err := setup.emailService.SendPasswordResetEmail(setup.testCtx, entities.User{
+		ID:    testUserId,
+		Email: "rob@test.com",
+	}, []common.UniformResourceIdentifier{testURI})
+	assert.NoError(t, err)
+}
+
+func Test_SendPasswordResetEmail__should_return_error_when_authorizer_returns_error(t *testing.T) {
+	setup := setupEmailTest(t)
+	defer setup.ctrl.Finish()
+	testURI, _ := common.NewURIFromString("test")
+
+	setup.mockTimeProvider.EXPECT().Now().Return(time.Unix(1, 0)).Times(1)
+	setup.mockAuthorizer.EXPECT().CreateServiceToken(setup.testCtx, testUserId, []common.UniformResourceIdentifier{testURI}, int64(1001)).
+		Return("", errors.New("authorizer err")).Times(1)
+
+	err := setup.emailService.SendEmailVerificationEmail(setup.testCtx, entities.User{
+		ID: testUserId,
+	}, []common.UniformResourceIdentifier{testURI})
+
+	assert.Error(t, err)
+}
+
+//
+//func Test_SendEmailVerificationEmail__should_return_error_when_authorizer_returns_error(t *testing.T) {
```

Comment:
Is there a reason why these tests are commented out?

Reply:
Sorry, forgot to remove them